File: src/CSharp/492.ConstructRectangle.cs

using System;
using System.Linq;
using System.Collections.Generic;

namespace PeefyLeetCode.ConstructRectangle
{
public class Solution 
{
    public int[] ConstructRectangle(int area) 
    {
        for (int i = (int)Math.Sqrt(area); i > 0; i--) 
        {
            if (area % i == 0)
                return new int[]{area / i, i};
        }
        return new int[]{area, 1};
    }
}

}

File: src/Java/492.ConstructRectangle.java

import java.awt.List;
import java.util.LinkedList;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.Comparator;
import java.util.HashMap;
import java.util.HashSet;

class Solution {
    public int[] constructRectangle(int area) {
        int sqrt = (int)Math.sqrt(area) + 1;
        for (int i = 0; i < sqrt; i++) {
            int j = sqrt - i - 1;
            if (area % j == 0){
                return new int[]{area / j, j};
            }
        }
        return new int[]{area, 1};
    }
}